src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -309,239 +347,447 @@ def test_WINNF_FT_S_MCP_1(self, config_filename):
     for iteration_content in config['iterationData']:
-
       # Execute steps for single iteration
-      self.executeSingleMCPIteration(test_type, iteration_content, sas_test_harness_objects,
-                                     domain_proxy_objects)
+      self.executeSingleMCPIteration(iteration_content)
 
     # Stopping Test harness servers
-    for test_harness in sas_test_harness_objects:
+    for test_harness in self.sas_test_harness_objects:
       test_harness.shutdown()
       del test_harness
 
-  def checkMcpIap(self, protected_entity, agg_interference_list, iap_refmodel_margin_list, delta_iap):
-    """Calculating IAP for each FSS and ESC sensor protection point,
-    and for at least 95% of the protection points of each PPA and GWPZ.
-    """
-    pass
-
-  def callAggregateInterferenceModel(self, cbsd_list, protected_entity):
-    """Args:
-         cbsd_list: List of cbsds with Authorized grants
-         protected_entity: A dictionary with multiple key-value pairs where
-         keys are protection entities
-
-       Returns:
-         Aggregate Interference list
-    """
-    pass
-
-  def executeSingleMCPIteration(self, test_type, iteration_content, sas_test_harness_objects,
-                                                          domain_proxy_objects):
+  def executeSingleMCPIteration(self, iteration_content):
     """Executes the steps from Step1 to Step22 for MCP and XPR testcases
 
     Args:
-       test_type: A string which indicates the testcase type
-       iteration_content: A dictionary with multiple key-value pairs that contain iteration data
-       sas_test_harness_objects: SAS Test Harnesses objects
-       domain_proxy_objects: Domain proxy objects
+      iteration_content: A dictionary with multiple key-value pairs that contain iteration data
     """
-    protected_entity_records = iteration_content['protectedEntities']
-    if 'fssRecords' in protected_entity_records:
-      for fss_record in protected_entity_records['fssRecords']:
+    # Step 5 : Inject IAP protected entities into UUT
+    self.protected_entity_records.extend(iteration_content['protectedEntities'])
+    if 'fssRecords' in self.protected_entity_records:
+      for fss_record in self.protected_entity_records['fssRecords']:
         self._sas_admin.InjectFss({'record': fss_record})
 
-    if 'gwpzRecords' in protected_entity_records:
-      for gwpz_record in protected_entity_records['gwpzRecords']:
+    if 'gwpzRecords' in self.protected_entity_records:
+      for gwpz_record in self.protected_entity_records['gwpzRecords']:
         self._sas_admin.InjectWisp({'record': gwpz_record})
+        # TODO: calculate and store the land category of the GWPZ
 
-    if 'escRecords' in protected_entity_records:
-      for esc_record in protected_entity_records['escRecords']:
+    if 'escRecords' in self.protected_entity_records:
+      for esc_record in self.protected_entity_records['escRecords']:
```

Comment:
run time error : “TypeError: list indices must be integers, not str”

Reply:
This was most likely an issue with how the protected_entity_records were being merged